### PR TITLE
automatic release to GitHub

### LINKED ISSUE #272: automatic release to GitHub
Current state: releases are build manually

Target state:
* Binaries are build for PRs, main branch and tags
* Binaries at tags are pushed to GitHub releases

### PR DESCRIPTION
**Pull request by ZyanKLee**
* disable update mechanism
* remove upx (can later be added again, if need be)
* generate dev binary and upload to pull_request

closes #272

### CODE REVIEW

**Comment by ZyanKLee** [comment]
For now I only implemented the dev build. What is missing is the actual releasing mechanics.

**Comment by ZyanKLee** [comment]
This PR is ready for review. Anyone?

**Review by TheDukeofErl — APPROVED** [review]
Had to take a little bit of time to familiarize myself with the Github CICD pipeline and it's integration. Looks good to me!

**Comment by Dadadah** [comment]
LGTM

**Comment by ZyanKLee** [comment]
It just dawned on me, that I totally forgot to implement the key handling for the signer script. Will do that later today - latest tomorrow.

**Review by ZyanKLee — COMMENTED** [review]
Now, this last commit adds the missing magic:

I created a secret variable in Github, which is decoded (`base64 -d`) and put in place. Then `make release` is run (which also runs the signer command) and finally I create a md5sum - just for good measures. In the end a release is being created and all those files are attached to it.

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 UPDATE_URL=
-UPDATE_PUBKEY=
+UPDATE_PUBKEY=Md2rdsS+b6W0trgcqa5lAWP978Zj0sFmubJ252OPKwc=
```
Comment (ZyanKLee) [inline Makefile:2]: This is the public key that corresponds with the new signer key. They are used to in the end let `update.go` verify that this is a sane artifact.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -19,12 +19,15 @@ jobs:
...
+          rm -rf ~/.config/noisetorch/
+          for f in bin/NoiseTorch_x64_*.tgz ; do md5sum ${f} | tee ${f}.md5sum ; done
       - name: Release
         uses: softprops/action-gh-release@v1
-        if: startsWith(github.ref, 'refs/tags/')
         with:
           files: |
             ${{ github.workspace }}/bin/NoiseTorch_x64_*.tgz
             ${{ github.workspace }}/bin/NoiseTorch_x64_*.tgz.sig
+            ${{ github.workspace }}/bin/NoiseTorch_x64_*.tgz.md5sum
```
Comment (ZyanKLee) [inline .github/workflows/release.yml:33]: md5sum's are more common and more people know how to use them to verify the integrity of files.

Comment (TheDukeofErl) [reply .github/workflows/release.yml:33]: Whoops, sorry, meant to do a comment, not a review, so I deleted that. I'm still partial to doing an sha256sum over an md5 or sha1, given the cheapness of those attacks. At the very least, could we include both md5 and sha256?

Comment (ZyanKLee) [reply .github/workflows/release.yml:33]: I would agree, if this was used to secure anything vital. But it's only a checksum to verify if the file has been altered or is another file altogether. You think this is still a reason to use some more sophisticated checksum algorithm?

Comment (ZyanKLee) [reply .github/workflows/release.yml:33]: oh, well. why not? -> see  536f5b3

Comment (TheDukeofErl) [reply .github/workflows/release.yml:33]: Thanks! In reality, there probably isn't a huge amount of risk but it's something that doesn't hurt at all to add :)

I appreciate it getting added to the PR.

**Review by TheDukeofErl — APPROVED** [review]

**Pull request merged**